pkg/dns/provider/changemodel.go

```diff
@@ -331,78 +377,54 @@ func (this *ChangeModel) Exec(apply bool, delete bool, name dns.DNSSetName, upda
 	oldset := view.dnssets[name]
 	newset := dns.NewDNSSet(name, spec.RoutingPolicy())
 	newset.UpdateGroup = updateGroup
-	newset.SetKind(spec.Kind())
 	if !delete {
-		this.ApplySpec(newset, oldset, p, spec)
+		this.ApplySpec(newset, p, spec)
 	}
 	mod := false
 	if oldset != nil {
-		this.Debugf("found old for %s %q", oldset.GetKind(), oldset.Name)
-		if this.IsForeign(oldset) {
-			err := &perrs.AlreadyBusyForOwner{Name: name, EntryCreatedAt: createdAt, Owner: oldset.GetOwner()}
-			retry := p.ReportZoneStateConflict(this.context.zone.getZone(), err)
-			if done != nil {
-				if apply && !retry {
-					done.SetInvalid(err)
+		this.Debugf("found old for entry %q", oldset.Name)
+		for ty, rset := range newset.Sets {
```

Comment:
The variable names have been chosen long time ago. Your new suggestion are not quite correct.
We could use `rtype` or `recordType` instead of `ty` and `recordSet` instead of `rset`.

Reply:
You're right, my bad! But I agree that we should prefer consistency over tiny name changes, let's keep it as it is.